hypervisor/dm/vpci/hostbridge.c

```diff
@@ -40,46 +40,46 @@
-	pci_vdev_write_cfg_u8(vdev, 0x48U, 0x1U);
-	pci_vdev_write_cfg_u8(vdev, 0x4aU, 0xd1U);
-	pci_vdev_write_cfg_u8(vdev, 0x4bU, 0xfeU);
-	pci_vdev_write_cfg_u8(vdev, 0x50U, 0xc1U);
-	pci_vdev_write_cfg_u8(vdev, 0x51U, 0x2U);
-	pci_vdev_write_cfg_u8(vdev, 0x54U, 0x33U);
-	pci_vdev_write_cfg_u8(vdev, 0x58U, 0x7U);
-	pci_vdev_write_cfg_u8(vdev, 0x5aU, 0xf0U);
-	pci_vdev_write_cfg_u8(vdev, 0x5bU, 0x7fU);
-	pci_vdev_write_cfg_u8(vdev, 0x60U, 0x1U);
-	pci_vdev_write_cfg_u8(vdev, 0x63U, 0xe0U);
-	pci_vdev_write_cfg_u8(vdev, 0xabU, 0x80U);
-	pci_vdev_write_cfg_u8(vdev, 0xacU, 0x2U);
-	pci_vdev_write_cfg_u8(vdev, 0xb0U, 0x1U);
-	pci_vdev_write_cfg_u8(vdev, 0xb3U, 0x7cU);
-	pci_vdev_write_cfg_u8(vdev, 0xb4U, 0x1U);
-	pci_vdev_write_cfg_u8(vdev, 0xb6U, 0x80U);
-	pci_vdev_write_cfg_u8(vdev, 0xb7U, 0x7bU);
-	pci_vdev_write_cfg_u8(vdev, 0xb8U, 0x1U);
-	pci_vdev_write_cfg_u8(vdev, 0xbbU, 0x7bU);
-	pci_vdev_write_cfg_u8(vdev, 0xbcU, 0x1U);
-	pci_vdev_write_cfg_u8(vdev, 0xbfU, 0x80U);
-	pci_vdev_write_cfg_u8(vdev, 0xe0U, 0x9U);
-	pci_vdev_write_cfg_u8(vdev, 0xe2U, 0xcU);
-	pci_vdev_write_cfg_u8(vdev, 0xe3U, 0x1U);
-	pci_vdev_write_cfg_u8(vdev, 0xf5U, 0xfU);
-	pci_vdev_write_cfg_u8(vdev, 0xf6U, 0x1cU);
-	pci_vdev_write_cfg_u8(vdev, 0xf7U, 0x1U);
+	pci_vdev_write_cfg_u8(vdev, PCIR_CLASS, (uint8_t)PCIC_BRIDGE);
+	pci_vdev_write_cfg_u8(vdev, PCIR_SUBCLASS, (uint8_t)PCIS_BRIDGE_HOST);
+
+	pci_vdev_write_cfg_u8(vdev, 0x34U, (uint8_t)0xe0U);
+	pci_vdev_write_cfg_u8(vdev, 0x3cU, (uint8_t)0xe0U);
+	pci_vdev_write_cfg_u8(vdev, 0x48U, (uint8_t)0x1U);
+	pci_vdev_write_cfg_u8(vdev, 0x4aU, (uint8_t)0xd1U);
+	pci_vdev_write_cfg_u8(vdev, 0x4bU, (uint8_t)0xfeU);
+	pci_vdev_write_cfg_u8(vdev, 0x50U, (uint8_t)0xc1U);
+	pci_vdev_write_cfg_u8(vdev, 0x51U, (uint8_t)0x2U);
+	pci_vdev_write_cfg_u8(vdev, 0x54U, (uint8_t)0x33U);
+	pci_vdev_write_cfg_u8(vdev, 0x58U, (uint8_t)0x7U);
+	pci_vdev_write_cfg_u8(vdev, 0x5aU, (uint8_t)0xf0U);
+	pci_vdev_write_cfg_u8(vdev, 0x5bU, (uint8_t)0x7fU);
+	pci_vdev_write_cfg_u8(vdev, 0x60U, (uint8_t)0x1U);
+	pci_vdev_write_cfg_u8(vdev, 0x63U, (uint8_t)0xe0U);
+	pci_vdev_write_cfg_u8(vdev, 0xabU, (uint8_t)0x80U);
+	pci_vdev_write_cfg_u8(vdev, 0xacU, (uint8_t)0x2U);
+	pci_vdev_write_cfg_u8(vdev, 0xb0U, (uint8_t)0x1U);
+	pci_vdev_write_cfg_u8(vdev, 0xb3U, (uint8_t)0x7cU);
+	pci_vdev_write_cfg_u8(vdev, 0xb4U, (uint8_t)0x1U);
+	pci_vdev_write_cfg_u8(vdev, 0xb6U, (uint8_t)0x80U);
+	pci_vdev_write_cfg_u8(vdev, 0xb7U, (uint8_t)0x7bU);
+	pci_vdev_write_cfg_u8(vdev, 0xb8U, (uint8_t)0x1U);
+	pci_vdev_write_cfg_u8(vdev, 0xbbU, (uint8_t)0x7bU);
+	pci_vdev_write_cfg_u8(vdev, 0xbcU, (uint8_t)0x1U);
+	pci_vdev_write_cfg_u8(vdev, 0xbfU, (uint8_t)0x80U);
+	pci_vdev_write_cfg_u8(vdev, 0xe0U, (uint8_t)0x9U);
+	pci_vdev_write_cfg_u8(vdev, 0xe2U, (uint8_t)0xcU);
+	pci_vdev_write_cfg_u8(vdev, 0xe3U, (uint8_t)0x1U);
+	pci_vdev_write_cfg_u8(vdev, 0xf5U, (uint8_t)0xfU);
+	pci_vdev_write_cfg_u8(vdev, 0xf6U, (uint8_t)0x1cU);
+	pci_vdev_write_cfg_u8(vdev, 0xf7U, (uint8_t)0x1U);
```

Comment:
Ｉ thought we'd better to use a static array to describe the configuration space field and value to reduce the code size and logic... Your thought?

Reply:
Definitely ... there are few minor cleanups to be done in hostbridge.c and pci_pt.c, mostly to replace magic numbers with macros from include/dm/pci.h. Do you prefer to commit these changes for now for MISRA-C violations, and do another cleanup patch later, or I can remove this part from this patch to avoid frequent change?

Reply:
It is fine to do later.